Assets/HotmaxEditorVR/propertiesMenu/propertiesMenu.cs

```diff
@@ -0,0 +1,262 @@
+    {
+        _selectedObject = value;
+
+        if (_selectedObject != null)
+        {
+            currentPage = 0;
+
+            //TODO: check to see if the animation class is attached
+            animatorText.text = "#no animation";
+
+            //--------------------------------------MATERIALS--------------------------------------\\
+            //before we add new materials clear out the list
+            materialNames.Clear();
+
+            //if the hit object has a renderer, get the material
+            if (_selectedObject.GetComponent<Renderer>())
+            {
+                //if there is more than one material loop and add
+                for (int i = 0; i < _selectedObject.GetComponent<Renderer>().materials.Length; i++)
+                {
+                    materialNames.Add(_selectedObject.GetComponent<Renderer>().materials[i].name);
+                }
+            }
+
+            //get all the materials in the children
+            //--TODO: Do this on GLTF serialization
+            Renderer[] renderers = _selectedObject.GetComponentsInChildren<Renderer>();
+            foreach (Renderer renderer in renderers)
+            {
+                materialNames.Add(renderer.material.name);
+            }
+
+            removeLastMaterialPage();
+
+            //get only unique instances of the material names
+            materialNames = materialNames.Distinct().ToList();
+
+            //create "chunks" groups of 4 to display the object
+            chunkedMaterialList = ListExtensions.ChunkBy(materialNames, 4);
+            showMaterialPage(0);
+
+            //--------------------------------------MASS--------------------------------------\\
+
+            //--GETTING
+            //if it has a Rb it should be under 90 mass units
+            if (_selectedObject.GetComponent<Rigidbody>())
+            {
+                selectedObjectRb = _selectedObject.GetComponent<Rigidbody>();
+
+                if (selectedObjectRb.mass < 60)
+                {
+                    physicsText.text = "#grabbable #throwable";
+                }
+                else if (selectedObjectRb.mass > 60 && selectedObjectRb.mass < 90)
+                {
+                    physicsText.text = "#heavy #non-grabbable";
+                }
+
+                slider.transform.localPosition = new Vector3(0, 0, selectedObjectRb.mass / 10);
+            }
+            else
+            {
+                slider.transform.localPosition = new Vector3(0, 0, 10);
+                physicsText.text = "#non-moveable";
+            }
+        }                    
+    }
+
+    //--Handles the menu buttons
+    void triggerClicked(object sender, ClickedEventArgs e)
+    {
+        triggerDown = true;
+
+        RaycastHit hit;
+        Ray ray = new Ray(inputManager.hand2.gameObject.transform.position, inputManager.hand2.gameObject.transform.forward);
+
+        if (Physics.Raycast(ray, out hit, Mathf.Infinity))
```

Comment:
i would do the same stuff here and move this to a iHittable interfaces

Reply:
done